Added initial structure

Golub-Sergey opened the pull request:
-Add setup.py 
-add new django app

Golub-Sergey committed:
added initial structure

Golub-Sergey committed:
added search form, custom tags for checking type of values of settings vars

Golub-Sergey committed:
add build status to .md

Golub-Sergey committed:
add tests, modify BaseView in views.py

Golub-Sergey committed:
refactored .md file

Golub-Sergey committed:
add docstring comments for all project

Golub-Sergey committed:
add docstring comments for all project

Golub-Sergey commented:
This app display settings of django project.

- add view with settings and search form for them
- add template for display view data
- add template tags for display tree of settings
- add view tests

flying-pi reviewed (changes requested):
Could you update the title of the pull request, because it contains whole functionality, not only initial structure?

flying-pi commented on README.md:
```diff
@@ -0,0 +1,34 @@
+    ] 
+to urls.py file in your django project directory. 
+
+## URL
```
Could you rename this section to something more meaningful? I.E. How to use.

flying-pi commented on outsettingsvalues/__init__.py:
```diff
@@ -0,0 +1,3 @@
+"""
+outsettingsvalues app settings file.
```
Seems it docstring relates to some particular file. Could you update it to reflect package content or in case, when we actually do not need this docstring - add some notation to ignore missing docstring?

flying-pi commented on outsettingsvalues/apps.py:
```diff
@@ -0,0 +1,15 @@
+"""
```
Why we need this module?

flying-pi commented on outsettingsvalues/templates/outsettingsvalues/index.html:
```diff
@@ -0,0 +1,79 @@
+            {% if item_val.items %}
+                <ul>
+                    {% checktype item_val as check %}
+                    {% if check == True %}
```
Could we transform equation for avoid comparison with `True`

flying-pi commented on outsettingsvalues/templatetags/__init__.py:
```diff
@@ -0,0 +1,3 @@
+"""
+outsettingsvalues templatetags file.
```
Seems it docstring relates to some particular file.

flying-pi commented on outsettingsvalues/templatetags/custom_filter.py:
```diff
@@ -0,0 +1,28 @@
+    """
+    Check type of args. Dict or not.
+
+    :param args:
```
Why we need an empty description of the function arguments?

flying-pi commented on outsettingsvalues/templatetags/custom_filter.py:
```diff
@@ -0,0 +1,28 @@
+    Check type of args. Dict or not.
+
+    :param args:
+    :return: True or False
```
For my point of view, here must be a description when the function returns true and when false. Or in case, when this description is trivial - remove it please.

flying-pi commented on outsettingsvalues/tests.py:
```diff
@@ -0,0 +1,82 @@
+        """
+        Create mane instances for all test.
+
+        :return: None
```
Do we really need documentation about function result (what it returns) here and below?

flying-pi commented on outsettingsvalues/tests.py:
```diff
@@ -0,0 +1,82 @@
+        """
+        Test status code about get request of BaseView url without param setting_name.
+
+        :return: True or False
```
Seems this function return None, could you clarify docstring or remove the part with return description completely in case, when it trivial?

flying-pi commented on outsettingsvalues/tests.py:
```diff
@@ -0,0 +1,82 @@
+from outsettingsvalues.views import BaseView
+
+
+class TestBaseView(TestCase):
```
Could you use mock object here to take control of the environment for the test(get expected result, which not depend on which platform it runs)

flying-pi commented on outsettingsvalues/views.py:
```diff
@@ -0,0 +1,48 @@
+                continue
+            if setting_name is None:
+                settings_list[i] = getattr(settings, i)
+            elif setting_name.upper() in i:
```
Could you take care of the case-independent search?

flying-pi commented on setup.py:
```diff
@@ -0,0 +1,22 @@
+    install_requires=[],
+    requires=[],
+    packages=["outsettingsvalues"],
+    description='Some description',
```
Could you clarify this description?

Masergik commented on outsettingsvalues/templates/outsettingsvalues/index.html:
```diff
@@ -0,0 +1,79 @@
+                </ul>
+            {% else %}
+                {% checktypelist item_val as check %}
+                {% if check == True %}
```
(Same to 24 string)
Can we use "if" without comparison with True?

Masergik commented on outsettingsvalues/templates/outsettingsvalues/index.html:
```diff
@@ -0,0 +1,79 @@
+                            <li>
+                                {% checktypelist i as checklist %}
+                                {% checktype i as checkdict %}
+                                {% if ckecklist == True %}
```
comparison with True

Masergik commented on outsettingsvalues/templates/outsettingsvalues/index.html:
```diff
@@ -0,0 +1,79 @@
+                                            <li>{{ j }}</li>
+                                        {% endfor %}
+                                    </ul>
+                                {% elif ckeckdict == True %}
```
comparison with True

Masergik commented on outsettingsvalues/templates/outsettingsvalues/index.html:
```diff
@@ -0,0 +1,79 @@
+                {% checktypelist item_val as check %}
+                {% if check == True %}
+                    <ul>
+                        {% for i in item_val %}
```
Maybe {% checktypelist i as checklist %} and {% checktype i as checkdict %} are better to insert after "for"?

Golub-Sergey committed:
- modify test_load_settings method from tests,py
- remove app.py,
- modify load_settings method from views.py
- modify methods from templatetags.custom_filter.py

Golub-Sergey commented:
- modify test_load_settings method from tests,py 
- remove app.py,
- modify load_settings method from views.py
- modify methods from templatetags.custom_filter.py
